File: eventpublisher/src/main/scala/tk/wangkexiong/osgi/scala/Activator.scala
package tk.wangkexiong.osgi.scala

import org.osgi.framework._

class Activator extends BundleActivator {

    def start( context: BundleContext ) {
         var bundleNames = context.getBundle().getHeaders
          
         println(bundleNames.get("Bundle-Name") + " Started ...")
     }

     def stop( context: BundleContext )  {
         var bundleNames = context.getBundle().getHeaders
          
         println(bundleNames.get("Bundle-Name") + " Stoped ...")
     }
}

File: eventconsumer/src/main/java/tk/wangkexiong/osgi/Activator.java
package tk.wangkexiong.osgi;

import org.osgi.framework.BundleActivator;
import org.osgi.framework.BundleContext;


public class Activator implements BundleActivator {
    @Override
    public void start(BundleContext context) throws Exception {
        System.out.println(getBundleName(context) + " Started ...");
    }

    @Override
    public void stop(BundleContext context) throws Exception {
        System.out.println(getBundleName(context) + " Stoped ...");
    }

    private String getBundleName(BundleContext context) {
        return context.getBundle().getHeaders().get("Bundle-Name");
    }
}
